test/tools/yulopti.cpp

```diff
@@ -278,7 +323,22 @@ Allowed options)",
 	string input;
 	try
 	{
-		input = readFileAsString(arguments["input-file"].as<string>());
+		if (arguments.count("input-file"))
+		{
+			string filename = arguments["input-file"].as<string>();
+			if (filename == "-")
+			{
+				if (!arguments.count("steps"))
+				{
+					cerr << options;
+					return 1;
+				}
```

Comment:
Actually, why not allow stdin in interactive mode too? Combined with #10275 this could allow us to pipe the compiler output directly into the tool and then work on it :)

Reply:
How would you tell at which point the input file ends and interaction input begins in stdin :-)?
I forced non-interactive mode, because that doesn't work trivially at least (we could of course wait for the ``}`` closing the main yul block that is expected in input anyways, then skip all whitespace and expect interactive input from there... but if we want anything like that we can do it separately :-)).

Reply:
Ah, you got me :) Seems obvious but somehow I didn't think of that. We can't read input from the user and pipe to stdin at the same time.